src/lookup.py

```diff
@@ -335,93 +335,140 @@ class EntityLookup(Lookup):
+                return number
 
-            def setBitValue(self, subtype, value):
-                subtype = bitSet(subtype, value, self.bitoffset, self.bits)
+            def setRawValue(self, number, value):
+                number = bitSet(number, value, self.offset, self.length)
 
-                return subtype
+                return number
 
-            def getValueFromIndex(self, index):
+            def getRawValueFromWidgetValue(self, widgetValue):
                 if self.dropdownvalues:
-                    return self.dropdownvalues[index]
+                    return self.dropdownvalues[widgetValue]
+
+                if self.widget == "dial":
+                    return (widgetValue - self.valueoffset) % (self.maximum + 1)
 
-                if self.display == "Dial":
-                    return (index - self.basevalue) % (self.maximum + 1)
+                if self.widget == "checkbox":
+                    return 1 if widgetValue else 0
 
-                return index - self.basevalue
+                return widgetValue - self.valueoffset
 
-            def getIndexedValue(self, subtype):
-                value = self.getBitValue(subtype)
+            def getWidgetValue(self, number, rawValue=None):
+                value = rawValue if rawValue is not None else self.getRawValue(number)
 
                 if self.dropdownvalues:
                     return self.dropdownvalues.index(value)
 
-                if self.display == "Dial":
-                    return (value + self.basevalue) % (self.maximum + 1)
+                if self.widget == "dial":
+                    return (value + self.valueoffset) % (self.maximum + 1)
 
-                return value + self.basevalue
+                if self.widget == "checkbox":
+                    return value == 1
 
-            def getDisplayValue(self, subtype, bitValue=None):
-                if bitValue is None:
-                    bitValue = self.getBitValue(subtype)
+                return value + self.valueoffset
 
-                if self.basevalue != 0 and self.display != "Dial":
-                    bitValue += self.basevalue
+            def getWidgetRange(self):
+                if self.widget == "dropdown":
+                    return len(self.dropdownvalues)
+                if self.widget == "dial":
+                    return self.minimum, self.maximum + 1
+
+                return self.minimum + self.valueoffset, self.maximum + self.valueoffset
+
+            def getDisplayValue(self, number, rawValue=None):
+                value = self.getWidgetValue(number, rawValue)
+
+                if self.unit == "Degrees":  # Match Isaac's degrees, where right is 0
+                    degrees = value * (360 / (self.maximum + 1))
+                    return (degrees + 90) % 360
 
                 if self.secondrange:
-                    bitValue = round(self.secondrange / bitValue / 30, 3)
+                    return round(self.secondrange / value / 30, 3)
```

Comment:
effectively, self.secondrange represents `round(self.secondrange / value, 3)`. Rather than calling this `secondrange`, let's call it `ScalarRange`. I'm also slightly concerned about the usability of this slider, since the values go 0, 5, 2.5, ... 0.3.

Reply:
Values are 5 - 0.312, does not start with 0. It's accurate precisely to the base game delay range, so I'm not sure how it could be better expressed. I'm up for changing it to ScalarRange and removing the built-in ` / 30` though, although that would result in an awkward magic number seeming value of 0.16667 for its only use case in BR.

Reply:
Okay actually I'm totally wrong about that "magic number" thing this is totally fine.